### PR TITLE
feat(gitpod): prebuilds + VS Code extensions

### PR DESCRIPTION
**Pull request by smacintyre**

### CODE REVIEW

**Comment by smacintyre** [comment]
@projen rebuild

File: src/gitpod.ts [hunk 1]
```diff
@@ -41,6 +43,54 @@ export interface GitpodDocker {
   readonly file?: string;
 }
 
+/**
+ * Configure the Gitpod App for prebuilds
+ * Currently the App only support GitHub.
+ * See: https://www.gitpod.io/docs/prebuilds/
+ */
+export interface GitpodPrebuildsConfig {
```
Comment (eladb) [inline src/gitpod.ts:51]: No need to use the word "Config" - everything is a config.

```suggestion
export interface GitpodPrebuilds {
```

File: src/gitpod.ts [hunk 2]
```diff
@@ -169,14 +225,37 @@ export interface GitpodOptions {
...
+  readonly prebuilds?: GitpodPrebuildsConfig;
+
+  /**
+   * Optional Include VS Code Extensions to be installed with the project
+   * See: https://www.gitpod.io/docs/vscode-extensions/
+   */
+  readonly extensions?: GitpodCodeExtensionId[];
 }
 
+type GitpodCodeExtensionsRender = {
```
Comment (eladb) [inline src/gitpod.ts:244]: Please move to the back of the file (these are non-public types).

File: src/gitpod.ts [hunk 3]
```diff
@@ -187,17 +266,24 @@ export class Gitpod extends Component {
...
+    }
+    if (options?.extensions) {
+      this.addExtensions(...options?.extensions);
+    }
 
     new YamlFile(this.project, GITPOD_FILE, {
       obj: {
         image: () => this.renderDockerImage(),
         tasks: () => this.renderTasks(),
+        github: () => this.renderPrebuilds(),
```
Comment (eladb) [inline src/gitpod.ts:280]: Is this really under the `github` field?

**Review by eladb — CHANGES_REQUESTED** [review]
looks good. a few notes.

File: src/gitpod.ts (hunk 3 above)
Comment (smacintyre) [reply src/gitpod.ts:280]: Yes, please see [the docs](https://www.gitpod.io/docs/prebuilds/#configure-the-github-app)
Only Github app can be configured through the YAML and has full support.

File: src/gitpod.ts (hunk 2 above)
Comment (smacintyre) [reply src/gitpod.ts:244]: You mean to the **end** of the file, right?

**Review by eladb — APPROVED** [review]

**Comment by smacintyre** [comment]
@projen rebuild

**Review by Chriscbr — COMMENTED** [review]
This looks alright - I can merge these changes into my PR afterwards.

I didn't know Gitpod expects extension IDs in a different format, so thanks for documenting that as well.

Was `.theia/settings.json` meant to be added? If so can we make it a managed file (by adding it thru `new JsonFile()` in .projenrc.js)?

File: .theia/settings.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (eladb) [inline .theia/settings.json:1]: Following up on @Chriscbr's question: this should be synthesized.

**Review by eladb — CHANGES_REQUESTED** [review]
See @Chriscbr's comment

**Comment by pgollucci** [comment]
agree.

**Pull request closed without merging**